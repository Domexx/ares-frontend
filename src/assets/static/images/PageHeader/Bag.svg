<svg width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M16.56 6C16.7244 6.117 16.9456 6.386 17.0111 6.557L17.0122 6.574L17.7633 17.995C17.7578 17.998 17.7522 18 17.7489 18.001H2.25444C2.25 18 2.24333 17.998 2.23667 17.995L2.98778 6.573L2.98889 6.556C3.05444 6.386 3.27444 6.117 3.43889 6H16.56ZM16.75 4H3.24889C2.00667 4 0.768889 5.563 0.768889 6.455L0 18.154C0 19.048 1.00667 20 2.24889 20H17.7489C18.9922 20 20 19.048 20 18.154L19.2311 6.455C19.2311 5.563 17.9922 4 16.75 4Z" fill="#B5B6B8"/>
<path d="M7.77776 8V3.893C7.77776 2.849 8.7211 2 9.8811 2H9.99999H10.06C11.22 2 12.2222 2.849 12.2222 3.893V8H14.4444V3.893C14.4444 1.743 12.5078 0 10.1189 0H9.8811C7.49221 0 5.55554 1.743 5.55554 3.893V8H7.77776Z" fill="#B5B6B8"/>
</svg>
