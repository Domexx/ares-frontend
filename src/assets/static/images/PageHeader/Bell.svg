<svg width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M19.5376 14.127C19.5121 14.109 17.0476 12.369 17.0476 6.607C17.0476 2.964 13.8864 0 10.0053 0C6.12411 0 2.96518 2.964 2.96518 6.607C2.96518 12.369 0.498478 14.109 0.489589 14.115C0.0840282 14.369 -0.093752 14.84 0.0484721 15.273C0.191807 15.707 0.622924 16.003 1.11182 16.003H18.8976C19.3787 16.003 19.7976 15.71 19.9454 15.284C20.0965 14.861 19.9254 14.389 19.5376 14.127ZM3.42963 13.939C4.30408 12.527 5.18965 10.21 5.18965 6.607C5.18965 4.102 7.35079 2.065 10.0053 2.065C12.662 2.065 14.822 4.103 14.822 6.607C14.822 10.21 15.7065 12.527 16.5809 13.94H3.42963V13.939ZM12.6875 17.146C12.0542 16.863 11.2508 17.031 10.8886 17.527C10.7486 17.718 10.5253 17.853 10.2586 17.91C9.9986 17.961 9.72749 17.933 9.49415 17.826C9.34082 17.756 9.21082 17.652 9.12082 17.529C8.7597 17.033 7.95413 16.861 7.32301 17.144C6.69189 17.427 6.47189 18.058 6.83412 18.552C7.1519 18.99 7.61191 19.355 8.1608 19.607C8.72748 19.867 9.35748 20 9.99749 20C10.3097 20 10.6242 19.969 10.9342 19.904C11.8853 19.707 12.682 19.228 13.1742 18.556C13.5364 18.061 13.3164 17.432 12.6875 17.146Z" fill="#B5B6B8"/>
</svg>
