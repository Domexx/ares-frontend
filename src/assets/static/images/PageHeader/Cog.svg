<svg width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M10 7C8.343 7 7 8.343 7 10C7 11.655 8.343 12.999 10 12.999C11.656 12.999 13 11.655 13 10C13 8.343 11.656 7 10 7ZM10 11C9.448 11 9 10.551 9 10C9 9.448 9.448 9 10 9C10.552 9 11 9.448 11 10C11 10.551 10.552 11 10 11ZM17.363 12.638C18.817 12.638 20 11.453 20 10C20 8.547 18.817 7.363 17.363 7.363H17.228C17.122 7.363 17.023 7.314 16.958 7.234C16.944 7.18 16.926 7.128 16.903 7.076C16.844 6.944 16.872 6.793 17.014 6.647C17.512 6.148 17.786 5.486 17.786 4.782C17.786 4.078 17.512 3.415 17.013 2.917C16.016 1.92 14.32 1.882 13.244 2.959C13.142 3.059 13.004 3.094 12.847 3.025C12.718 2.97 12.635 2.845 12.635 2.634C12.637 1.182 11.454 0 10 0C8.546 0 7.363 1.182 7.363 2.636V2.771C7.363 2.877 7.314 2.977 7.234 3.042C7.181 3.056 7.128 3.073 7.077 3.096C6.949 3.154 6.813 3.148 6.647 2.983C5.651 1.989 3.916 1.988 2.917 2.986C1.891 4.014 1.892 5.687 2.961 6.756C3.062 6.858 3.089 7.01 3.007 7.2C2.957 7.336 2.831 7.427 2.636 7.429C1.183 7.429 0 8.612 0 10.065C0 11.519 1.183 12.702 2.636 12.702H2.774C2.912 12.702 3.037 12.785 3.096 12.924C3.154 13.056 3.126 13.208 2.983 13.353C2.486 13.851 2.212 14.514 2.212 15.218C2.213 15.923 2.488 16.585 2.985 17.081C3.981 18.077 5.674 18.119 6.756 17.039C6.857 16.94 6.898 16.866 7.199 16.993C7.335 17.042 7.425 17.169 7.428 17.363C7.428 18.817 8.611 20 10.065 20C11.519 20 12.702 18.817 12.702 17.363V17.226C12.702 17.087 12.785 16.963 12.924 16.904C13.053 16.846 13.187 16.853 13.352 17.016C14.351 18.011 16.087 18.012 17.082 17.014C17.58 16.515 17.853 15.852 17.853 15.148C17.852 14.444 17.578 13.782 17.04 13.245C16.939 13.141 16.911 12.989 16.976 12.847C17.029 12.72 17.154 12.638 17.363 12.638ZM15.666 14.7C15.787 14.82 15.854 14.979 15.854 15.149C15.854 15.319 15.787 15.479 15.666 15.602C15.427 15.843 15.061 15.897 14.71 15.544C14.033 14.882 12.968 14.699 12.128 15.069C11.264 15.438 10.705 16.283 10.703 17.222V17.362C10.703 17.714 10.417 17.998 10.066 17.998C9.715 17.998 9.429 17.714 9.429 17.265C9.406 16.295 8.802 15.45 7.95 15.14C7.649 15.007 7.329 14.939 7 14.939C6.382 14.939 5.798 15.177 5.3 15.667C5.06 15.907 4.64 15.908 4.399 15.665C4.151 15.419 4.151 15.015 4.457 14.708C5.132 14.018 5.316 12.999 4.931 12.128C4.563 11.264 3.717 10.704 2.777 10.702H2.637C2.285 10.702 2 10.416 2 10.065C2 9.714 2.285 9.428 2.733 9.428C3.704 9.405 4.548 8.802 4.86 7.948C5.25 7.064 5.065 6.048 4.333 5.3C4.085 5.052 4.085 4.647 4.334 4.398C4.574 4.158 4.937 4.1 5.291 4.456C5.92 5.07 6.871 5.282 7.694 5C7.778 4.987 7.861 4.964 7.939 4.93C8.8 4.56 9.359 3.716 9.363 2.775V2.635C9.363 2.285 9.649 2 10 2C10.351 2 10.637 2.285 10.637 2.712C10.639 3.649 11.198 4.494 12.051 4.86C12.916 5.24 13.935 5.082 14.7 4.333C14.941 4.093 15.356 4.09 15.601 4.334C15.721 4.454 15.788 4.613 15.788 4.783C15.788 4.952 15.722 5.114 15.546 5.29C14.917 5.932 14.712 6.856 15.001 7.695C15.014 7.778 15.036 7.86 15.07 7.939C15.439 8.801 16.283 9.36 17.224 9.364H17.363C17.715 9.364 18 9.649 18 10C18 10.351 17.715 10.637 17.288 10.637C16.35 10.639 15.505 11.198 15.142 12.048C14.749 12.933 14.933 13.951 15.666 14.7Z" fill="#B5B6B8"/>
</svg>
