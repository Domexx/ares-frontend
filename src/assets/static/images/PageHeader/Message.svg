<svg width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M12 2C13.103 2 14 2.897 14 4V9C14 10.103 13.103 11 12 11H8H7.754L7.515 11.06L2 12.438V9V4C2 2.897 2.897 2 4 2H12ZM12 0H4C1.791 0 0 1.791 0 4V9V15L8 13H12C14.209 13 16 11.209 16 9V4C16 1.791 14.209 0 12 0Z" fill="#B5B6B8"/>
<path d="M12 5.012C12 5.558 11.558 6 11.013 6H4.987C4.442 6 4 5.558 4 5.012V4.988C4 4.442 4.442 4 4.987 4H11.012C11.558 4 12 4.442 12 4.988V5.012Z" fill="#B5B6B8"/>
<path d="M10 8.047C10 8.574 9.573 9 9.047 9H4.953C4.427 9 4 8.574 4 8.047V7.953C4 7.426 4.427 7 4.953 7H9.047C9.573 7 10 7.426 10 7.953V8.047Z" fill="#B5B6B8"/>
<path d="M17 5.14197V7.27797C17.595 7.62497 18 8.26297 18 8.99997V14V17.438L12.484 16.059L12.246 16H12H8.00003C7.26303 16 6.62503 15.595 6.27803 15H4.14203C4.58803 16.722 6.13903 18 8.00003 18H12L20 20V14V8.99997C20 7.13897 18.722 5.58797 17 5.14197Z" fill="#B5B6B8"/>
</svg>
