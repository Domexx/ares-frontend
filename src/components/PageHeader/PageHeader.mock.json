{
  "main": [
    {
      "id": "cosmic_shop",
      "name": "Shop",
      "url": "shop",
      "icon": "bag"
    },
    {
      "id": "cosmic_friends",
      "name": "Friends",
      "url": "friends",
      "icon": "smile"
    },
    {
      "id": "cosmic_messages",
      "name": "Messages",
      "url": "messages",
      "icon": "messages"
    },
    {
      "id": "cosmic_important_news",
      "name": "Important",
      "url": "important-news",
      "icon": "bell"
    }
  ],
  "secondary": [
    {
      "id": "cosmic_settings",
      "name": "Settings",
      "url": "settings",
      "icon": "cog"
    }
  ]
}
