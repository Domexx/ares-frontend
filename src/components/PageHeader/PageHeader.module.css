/* === === === === PageHeader === === === === */

.PageHeader {
  position: sticky;
  width: 100%;
}

.PageHeader-top {
  display: flex;
  align-items: center;
  justify-content: space-between;
  width: 100%;
  height: 80px;
  padding: 0 50px 0 30px;
  background: var(--clr-dark-600);
}

.PageHeader-topLeft {
  display: flex;
  align-items: center;
}

.PageHeader-topLeft-button {
  width: 60px;
  height: 57px;
  padding: 10px;
  border: 2px solid transparent;
  border-radius: 8px;
  background: transparent;
  cursor: pointer;
  transform: rotate(0deg);
  transition: 0.25s ease-in-out;
  appearance: none;
}

.PageHeader-topLeft-button:hover {
  border: 2px solid var(--clr-clickable);
  background: var(--clr-dark-300);
  outline: 0;
}

.PageHeader-topLeft-button:active,
.PageHeader-topLeft-button:focus {
  outline: 0;
}

.PageHeader-topLeft-button span {
  display: block;
  width: 100%;
  height: 4px;
  border-radius: 4px;
  background-color: #ffffff;
}

.PageHeader-topLeft-button span + span {
  margin-top: 7px;
}

.PageHeader-logo {
  right: 3px;
  margin-top: 10px;
  scale: 0.75;
}

.PageHeader-search {
  position: relative;
  min-width: 500px;
  margin-left: 85px;
  padding: 0;
  border-radius: 12px;
}

.PageHeader-searchField {
  width: 100%;
  padding: 15px 70px 15px 20px;
  border: 2px solid var(--clr-dark-400);
  border-radius: 12px;
  background: var(--clr-dark-400);
  color: #ffffff;
  transition: border 250ms ease-in-out;
  appearance: none;
}

.PageHeader-searchField:focus,
.PageHeader-searchField:active {
  border: 2px solid var(--clr-clickable);
  outline: 0;
}

.PageHeader-searchButton {
  position: absolute;
  top: 50%;
  right: 20px;
  display: flex;
  align-items: center;
  justify-content: center;
  padding: 10px;
  border: 2px solid transparent;
  border-radius: 8px;
  background: transparent;
  transform: translateY(-50%);
  transition: background 250ms ease-in-out, border 250ms ease-in-out;
}

.PageHeader-searchButton:focus,
.PageHeader-searchButton:active {
  outline: 0;
}

.PageHeader-searchButton:focus,
.PageHeader-searchButton:active,
.PageHeader-searchButton:hover {
  border: 2px solid var(--clr-clickable);
  background: var(--clr-dark-300);
}

@media (--laptop) {
  .PageHeader-logo {
    display: none;
  }
}

@media (--tablet) {
  .PageHeader-topLeft {
    flex: 1;
  }

  .PageHeader-search {
    width: 100%;
    min-width: unset;
    margin-right: 40px;
    margin-left: 40px;
  }
}

@media (--phone) {
  .PageHeader-search {
    display: none;
  }
}
