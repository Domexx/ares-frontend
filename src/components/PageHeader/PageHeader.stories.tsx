import { Story } from '@storybook/react/types-6-0';
import React, { ComponentProps } from 'react';
import PageHeaderComponent, { Props } from './PageHeader';

import PageHeaderMock from './PageHeader.mock.json';

const Component = {
  title: 'Components/Page Header',
  component: PageHeaderComponent,
  argTypes: {},
};

const Template: Story<ComponentProps<typeof PageHeaderComponent>> = (args) => {
  return <PageHeaderComponent {...args} />;
};

export const PageHeader = Template.bind({});
export default Component;

const { main, secondary } = PageHeaderMock;

// https://storybook.js.org/docs/react/essentials/controls
PageHeader.args = {
  items: {
    main,
    secondary,
  },
} as Props;
