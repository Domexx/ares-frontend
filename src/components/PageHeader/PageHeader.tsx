import React from 'react';
import clsx from 'clsx';
import { useFormik } from 'formik';
import { useMenuContext } from '../../shared/contexts/MenuContext';
import { PageHeaderNav } from '../../shared/interfaces/PageHeaderItem';
import Logo from '../../elements/Logo';

import { ReactComponent as SearchIcon } from '../../assets/static/images/Search.svg';

import s from './PageHeader.module.css';
import List from './List';

export type Props = {
  items: PageHeaderNav;
};

const PageHeader: React.FC<Props> = ({ items }) => {
  const { menuOpen, setMenuOpen } = useMenuContext();
  const burgerClasses = clsx(s['PageHeader-topLeft-button'], menuOpen ? s['is-Active'] : '');

  const searchForm = useFormik({
    initialValues: {
      search: '',
    },
    onSubmit: (values) => {
      console.info(values);
    },
  });

  const changeMenu = () => {
    setMenuOpen(!menuOpen);
  };

  return (
    <header className={`${s['PageHeader']} u-grid`}>
      <div className={`${s['PageHeader-topLeft']} u-grid`}>
        <button className={burgerClasses} type="button" onClick={changeMenu}>
          <span />
          <span />
          <span />
        </button>
        <div className={s['PageHeader-logo']}>
          <Logo />
        </div>
        <form onSubmit={searchForm.handleSubmit} className={s['PageHeader-search']}>
          <input
            {...searchForm.getFieldProps('search')}
            className={s['PageHeader-searchField']}
            type="text"
            name="search"
            placeholder="Suche hier nach Usern oder Gruppen..."
          />
          <button className={s['PageHeader-searchButton']} aria-label="Label" type="submit">
            <SearchIcon />
          </button>
        </form>
      </div>
      <div className={`${s['PageHeader-topRight']} u-grid`}>
        <List main={items.main} secondary={items.secondary} />
      </div>
    </header>
  );
};

export default PageHeader;
