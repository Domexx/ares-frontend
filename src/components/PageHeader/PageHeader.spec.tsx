import React from 'react';

import { render } from '@testing-library/react';

import PageHeader from './PageHeader';

describe('Page Header Component', () => {
  it('should render properly', () => {
    const { container } = render(<PageHeader />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
