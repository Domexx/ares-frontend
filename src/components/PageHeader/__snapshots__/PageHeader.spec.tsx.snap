// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Header Component should render properly 1`] = `
<header
  class="PageHeader u-grid"
>
  <div
    class="PageHeader-topLeft u-grid"
  >
    <button
      class="PageHeader-topLeft-button is-Active"
      type="button"
    >
      <span />
      <span />
      <span />
    </button>
    <div
      class="PageHeader-logo"
    />
    <form
      class="PageHeader-search"
    >
      <input
        class="PageHeader-searchField"
        name="search"
        placeholder="Suche hier nach Usern oder Gruppen..."
        type="text"
        value=""
      />
      <button
        aria-label="Label"
        class="PageHeader-searchButton"
        type="submit"
      >
        <svg />
      </button>
    </form>
  </div>
  <div
    class="PageHeader-topRight u-grid"
  />
</header>
`;
